pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -88,13 +92,35 @@ func (r *Registration) syncNode(ctx context.Context, nodeClaim *v1beta1.NodeClai
 	// Sync all taints inside NodeClaim into the Node taints
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.Taints)
 	node.Spec.Taints = scheduling.Taints(node.Spec.Taints).Merge(nodeClaim.Spec.StartupTaints)
+	// Remove karpenter.sh/unregistered taint
+	taints, err := isolateKarpenterStartupTaint(node.Spec.Taints)
+	if err != nil {
+		return fmt.Errorf("missing required startup taint: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("missing required startup taint, %w", err)
```

nit: style, prefer using a comma on error wrapping